Fastmail.md setup Update

mtalexan opened the pull request:
Fastmail supports a ton of domains that a user can pick during sign up that don't have their own IMAP or SMTP gateways. This requires using the Advanced setup as part of the "common" use case.
The app-specific password generation combined with the webmail email address as the "username" for authentication easily causes confusion for custom domains about which fields during setup are for the custom domain and which are for the gateway login names (especially the IMAP password).

r10s reviewed (approved):
thanks for the detailed work!

maybe we could, in a subsequent pr, also add the ~100 domains to the list of domains.

this would be quite some list, however, we did similar things already for other large providers (web.de with ~30 domains and posteo with ~50 domains). i think for large providers, this effort is reasonable.  
maybe we could also detect fastmail somehow by dns, that would also be great for custom-domains, but we are not there yet :)

mtalexan commented:
Do you mean filing each domain as a separate provider in the documentation? Fastmail doesn't run those websites as white label of their service, they all actually direct back to the Fastmail.com site. Being email though, they clearly have the MX records configured at them.  I'm not sure it's worth having separate documentation for each domain in that case when users would still be looking up the company name (Fastmail) as their provider.  
If you mean in the provider database in terms of automaticly detecting the provider during the configuration without the user needing to fill in the Advanced settings, that's certainly doable, though I'm not familiar with the code myself. Fastmail regularly adds to their domain list though, so the instructions would probably still need to include info in case they added a new domain and someone tried to use it before it was added here.

Fully automatic detection for all domains might be possible, but I think there are some potential technical and usability issues. While it's always possible to do the MX record lookup on the domain a user lists in their email, otherwise email to them wouldn't ever work, I don't know if that can be directly translated into an IMAP and SMTP server. Certainly it can be guessed that they'll probably be `imap.domain.tld` and `smtp.domain.tld` for the `domain.tld` listed in the MX record at their email domain name, but that's merely a common convention and isn't always true. Furthermore, in cases like this especially, the Login Name for the IMAP and SMTP servers may be entirely different from the email address (common in corporate environments where an SSO username may be used instead). Even if the technical issues are overcome, would the user want it to automatically make these many assumptions without telling them? For well known services where the email domain is specifically known to go to the separate hosting domain, certainly, but in the generic case probably not. I know I would prefer my custom domain name lookup not go to some arbitrary host and send it my credentials if I accidentally typo something.  
At most I think the Advanced settings values could be prepopulated with guesses based on MX domain lookups, and subdomain guesses that are probed for existence and protocol support confirmation, but would still require the user to confirm before being used.

mtalexan commented:
I suppose it helps if I read the README on this repo more closely (*facepalm*).

I can certainly expand the list of providers to add all the ones Fastmail supports. Should I update the date from `01-2020` as well?

mtalexan commented:
Ok, just added all the Fastmail domains to the YAML.  I updated the instructions accordingly, but still reference the Advanced if a user is trying to use one of the other domains as an alias (allowed by Fastmail).

hpk42 commented:
On Sun, Jan 30, 2022 at 10:47 -0800, Mike wrote:
> I can certainly expand the list of providers to add all the ones Fastmail supports. Should I update the date from `01-2020` as well?

sure!
